src/core/mount.c

```diff
@@ -1750,8 +1752,6 @@ static int mount_load_proc_self_mountinfo(Manager *m, bool set_flags) {
                 if (!device || !path)
                         continue;
 
-                device_found_node(m, device, DEVICE_FOUND_MOUNT, DEVICE_FOUND_MOUNT);
```

Comment:
i can't really believe this would be so slow. checking that the specified device node already has a unit is an O(1) hash table lookup. Is that supposed to be so slow? what am i missing?